### PR TITLE
fix(infura): allow DID reg to take in Infura config

### PR DESCRIPTION
**Pull request by whitneypurdum**
### Description

Allow DID Registry to take in Infura config
### Contributor checklist

- [x] Breaking changes - check for any existing interfaces changes that are not backward compatible, removed method etc.
- [x] Documentation - document your code, add comments for method, remember to check if auto generated docs were updated.
- [x] Tests - add new or updated existed test for changes you made.
- [x] Migration guide - add migration guide for every breaking change.
- [x] Configuration correctness - check that any configuration changes are correct ex. default URLs, chain ids, smart contract verification on [Volta explorer](https://volta-explorer.energyweb.org/) or [EWC explorer](https://explorer.energyweb.org/).

### CODE REVIEW

File: e2e/cache-client.e2e.ts [hunk 1]
```diff
@@ -18,7 +18,22 @@ describe.skip('Cache client', () => {
     );
     const { connectToDidRegistry, assetsService } =
       await connectToCacheServer();
-    const { claimsService, didRegistry } = await connectToDidRegistry();
+    const projectId = '2DCe3TNAHzzmoSBrcPyWx4NOyrv';
```
Comment (whitneypurdum) [inline e2e/cache-client.e2e.ts:21]: I will change these configs - what configs should we be using for test files? I will make helper and share among e2e tests.

File: src/modules/did-registry/did.types.ts [hunk 2]
```diff
@@ -12,6 +12,13 @@ export interface AssetProfile {
   icon?: string;
 }
 
+export interface IpfsConfig {
```
Comment (whitneypurdum) [inline src/modules/did-registry/did.types.ts:15]: I think this is the best file for this interface, but have doubts

**Comment by whitneypurdum** [comment]
I am not running the tests in case it is not good to use the config that's hard coded.

File: e2e/cache-client.e2e.ts (hunk 1 above)
Comment (jrhender) [reply e2e/cache-client.e2e.ts:21]: Best to use
ProjectID: 2DD5GcJRpMeuqMoKsgT95AQ8OI3
ProjectSecret: 842995992aa1edd763196b30d727a45d

**Comment by jrhender** [comment]
> I am not running the tests in case it is not good to use the config that's hard coded.

I think ideally, it would be possible to run the tests against a mock IPFS `ipfsDaemon`.
But we can use actually project values for this first PR if that is easier.

File: src/init.ts [hunk 3]
```diff
@@ -98,13 +98,13 @@ export async function init(signerService: SignerService) {
     );
 
     async function connectToDidRegistry(
-      ipfsStore?: string
+      ipfsConfig: IpfsConfig
```
Comment (whitneypurdum) [inline src/init.ts:101]: Is there a reason why this should stay optional? Should there be a default config set in ICL in case it is not passed in?

**Review by jrhender — APPROVED** [review]

**Review by nichonien — APPROVED** [review]

**Pull request merged**